tools/ts-generator/rpc/rpc_gen_js.py

```diff
@@ -79,72 +82,135 @@
     RpcHeader.toBytes(rpc, reserve);
 }
 
+const startReadRequest = (fn: ApplyFn) => (socket: Socket): void => {
+  socket.once("data", (data) => readRequestBuffer(data, fn, socket));
+};
+
+/**
+ * given a buffer with request, it read a rpc header and rpc header payload size
+ * if the buffer has a complete payload size it return buffer, Otherwise, it
+ * waits for n "data" events with necessary data for complete payload size.
+ * @param buffer
+ * @param fn
+ * @param socket
+ */
+const readRequestBuffer = (buffer: Buffer, fn: ApplyFn, socket: Socket) => {
+  const [rpcHeader, crcValidation, crc32] = validateRpcHeader(
+    buffer.slice(0, rpcHeaderSize)
+  );
+  if (!crcValidation) {
+    throw (
+      `Crc32 inconsistent, expect: ${rpcHeader.headerChecksum} ` +
+      `generated: ${crc32}`
+    );
+  } else {
```

Comment:
This else isn't needed, since the if block's content always throws.

Reply:
Got it.